### PR TITLE
solves #4 - initial structure

### PR DESCRIPTION
**Pull request by jenriquejr**
connects #4

### CODE REVIEW

**Comment by jenriquejr** [comment]
this is still work in progress

File: src/app/router/server-router.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    get: function _respond() {
+      this.res.writeHead(200, { 'Content-Type': 'text/html' })
+      this.res.end(
+        React.renderToStaticMarkup(
+          Page(null, {__html: React.renderToString(App({}))})
+        )
+      );
+    }
+  };
+}
```
Comment (jenriquejr) [inline src/app/router/server-router.js]: this handler block does not belongs here

File: src/app/server.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+var server = http.createServer(function (req, res) {
+  router.dispatch(req, res, function (err) {
+    if (err) {
+      res.writeHead(404);
+      res.end();
+    }
+
+    console.log('[' + res.statusCode + ' - ' + res.statusMessage + '] ' + req.url);
```
Comment (jenriquejr) [inline src/app/server.js]: verify what do we want to log

File: src/app/server.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  });
+});
+
+
+server.listen(8000);
```
Comment (jenriquejr) [inline src/app/server.js]: this file `server.js` might be moved outside of `app` directory

**Comment by jenriquejr** [comment]
@eespinal @kdemoya @jsainthilaire want to start getting feedback if possible

File: src/app/components/App.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+var App = React.createClass({
+
+  render: function () {
+
+    return (
+      <div>this is something different</div>
+    );
+  }
+
+});
```
Comment (kdemoya) [inline src/app/components/App.jsx]: Should this become the following? 

``` javascript
class App extends React.Component {
  ...
}
```

Comment (jenriquejr) [inline src/app/components/App.jsx]: that's correct! willing to do that already

File: src/app/templates/MainTemplate.jsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+var React =  require('react');
+var Head = require('./partials/Head.jsx');
+var Scripts = require('./partials/Scripts.jsx');
+
+
```
Comment (kdemoya) [inline src/app/templates/MainTemplate.jsx:11]: We want to define `propTypes`, `defaultProps`, etc... outside of the class, as following:

``` javascript
var propTypes = {
  children: React.PropTypes.shape({
    __html: React.PropTypes.string.isRequired
  }).isRequired
}
```

...

File: src/app/templates/MainTemplate.jsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        <body>
+          <div id="react-app" dangerouslySetInnerHTML={this.props.children} />
+          <Scripts />
+        </body>
+      </html>
+    );
+  }
+
+});
+
```
Comment (kdemoya) [inline src/app/templates/MainTemplate.jsx:28]: ...And then add them to the class like this:

``` javascript
MainTemplate.propTypes = propTypes;
```

**Comment by kdemoya** [comment]
@jenriquejr Ideally we want to use `setup/app-base#4` instead of `setup#4/app-base` as branch naming convention so Waffle can keep track automatically.

**Comment by jenriquejr** [comment]
@kdemoya does waffle keeps track of that branch name even though it is on a different repo?

**Comment by kdemoya** [comment]
@jenriquejr Once it sees a PR against the original repo, it should.

File: src/app/components/App.jsx (hunk 4 above)
Comment (jsainthilaire) [inline src/app/components/App.jsx]: If we are using this syntax, should we use "import React from 'react';" when requiring a module? @jenriquejr @kdemoya

Comment (kdemoya) [inline src/app/components/App.jsx]: @jsainthilaire Yes! we can even do:

``` javascript
import React, { Component, PropTypes } from 'react';
```

and then:

``` javascript
class App extends Component {
  ...
}
```

File: src/app/templates/MainTemplate.jsx (hunk 5 above)
Comment (kdemoya) [inline src/app/templates/MainTemplate.jsx:11]: We can also (and probably should) use `const` instead of `var`.

**Comment by jenriquejr** [comment]
@kdemoya I just verified that this naming convention works fine with waffle - 
it did not update the status automatically because the branch belongs to a diffrent repo

**Comment by jenriquejr** [comment]
this is updated but still work in progress. feel free to clone and play around.
let me know how you feel about this.

**Comment by kdemoya** [comment]
@jenriquejr I see... seems like we will need to use the `connect` keyword then.

File: src/server/handlers/app.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+const App = React.createFactory(require('../../app/components/App'));
+const Page = React.createFactory(require('../../app/templates/MainTemplate'));
+
+
+function appKickOff() {
+  let app = React.renderToString(App({}));
+  let page = React.renderToStaticMarkup(Page(null, {__html: app}));
```
Comment (kdemoya) [inline src/server/handlers/app.js:8]: :+1: for `let`

**Comment by kdemoya** [comment]
This is a great start, let's hold until we make a decision in https://github.com/koding4food/reactive-gram/issues/10 to make the changes accordingly and then merge.

**Comment by jenriquejr** [comment]
@kdemoya @jsainthilaire I just removed all sass initial implementation,
since we still need to decide how are we going to do style. See #10 for reference

**Comment by kdemoya** [comment]
@jenriquejr This is not part of your changes, but could you please remove SASS from our `.gitignore` file?

**Comment by jenriquejr** [comment]
@kdemoya sorry, saw it too late

**Pull request merged**